src/java/test/org/apache/zookeeper/server/quorum/QuorumPeerConfigTest.java

```diff
@@ -103,6 +103,23 @@ public void testCustomSSLAuth()
         }
     }
 
+    /**
+     * Test case for https://issues.apache.org/jira/browse/ZOOKEEPER-2873
+     */
+    @Test
+    public void testSamePortConfiguredForClientAndElection() throws IOException, ConfigException {
+        QuorumPeerConfig quorumPeerConfig = new QuorumPeerConfig();
+        try {
+            Properties zkProp = getDefaultZKProperties();
+            zkProp.setProperty("server.1", "localhost:2888:2888");
+            quorumPeerConfig.parseProperties(zkProp);
+            fail("ConfigException is expected");
+        } catch (ConfigException ce) {
```

Comment:
Instead of the try-catch paradigm, can we use `@Test(expected = ConfigException.class)`?

Reply:
I agree that your test as written is more exact, my proposal is more about future proofing the test so it doesn't get hung up on the exact language of the exception thrown.

I will happily defer the point to someone with more ZooKeeper experience about which is more in the project's style.

Reply:
You convinced me. I'm not fun of catching exception in unit tests either, I took the inspiration from the other tests in the class :)

Thanks for bringing the attention to it!